packages/gatsby/src/redux/reducers/pending-page-data-writes.ts

```diff
@@ -0,0 +1,28 @@
+import { ActionsUnion, IGatsbyState } from "../types"
+
+export const pendingPageDataWritesReducer = (
+  state: IGatsbyState["pendingPageDataWrites"] = {
+    pagePaths: new Set(),
+    templatePaths: new Set(),
+  },
+  action: ActionsUnion
+): IGatsbyState["pendingPageDataWrites"] => {
+  switch (action.type) {
+    case `ADD_PENDING_PAGE_DATA_WRITE`:
+      state.pagePaths.add(action.payload.path)
+      return state
+
+    case `ADD_PENDING_TEMPLATE_DATA_WRITE`:
```

Comment:
Just a note - in this PR - this case is never hit (we never emit `ADD_PENDING_TEMPLATE_DATA_WRITE` here yet). This is just preparation for follow up PRs that will make use of it (we will emit this whenever webpack's dependency graph changes will cause change to which static queries are assigned to particular page template)